sda-commons-starter/src/main/java/org/sdase/commons/starter/SdaPlatformBundle.java

```diff
@@ -0,0 +1,352 @@
+    }
+
+    @Override
+    public ConsumerTokenConfigBuilder<C> withoutCorsSupport() {
+      return this;
+    }
+
+    @Override
+    public ConsumerTokenConfigBuilder<C> withCorsConfigProvider(
+        CorsConfigProvider<C> corsConfigProvider) {
+      this.corsBundleBuilder = CorsBundle.builder().withCorsConfigProvider(corsConfigProvider);
+      return this;
+    }
+
+    @Override
+    public OpenApiInitialBuilder<C> withoutConsumerTokenSupport() {
+      return this;
+    }
+
+    @Override
+    public OpenApiInitialBuilder<C> withOptionalConsumerToken() {
+      consumerTokenConfig = new ConsumerTokenConfig();
+      consumerTokenConfig.setOptional(true);
+      this.consumerTokenBundleBuilder =
+          ConsumerTokenBundle.builder().withConfigProvider(c -> consumerTokenConfig);
+      return this;
+    }
+
+    @Override
+    public ConsumerTokenRequiredConfigInitialBuilder<C> withRequiredConsumerToken() {
+      consumerTokenConfig = new ConsumerTokenConfig();
+      consumerTokenConfig.setOptional(false);
+      this.consumerTokenBundleBuilder =
+          ConsumerTokenBundle.builder().withConfigProvider(c -> consumerTokenConfig);
+      return this;
+    }
+
+    @Override
+    public OpenApiInitialBuilder<C> withConsumerTokenConfigProvider(
+        ConsumerTokenConfigProvider<C> consumerTokenConfigProvider) {
+      this.consumerTokenBundleBuilder =
+          ConsumerTokenBundle.builder().withConfigProvider(consumerTokenConfigProvider);
+      return this;
+    }
+
+    @Override
+    public OpenApiInitialBuilder<C> withExcludePatternsForRequiredConsumerToken(String... regex) {
+      if (this.consumerTokenConfig == null) {
+        throw new IllegalStateException(
+            "ConsumerToken support can't be configured because it is disabled.");
+      }
+      this.consumerTokenConfig.getExcludePatterns().addAll(Arrays.asList(regex));
+      return this;
+    }
+
+    // PlatformBundleBuilder contains programmatic configuration
+
+    @Override
+    public PlatformBundleBuilder<C> disableBufferLimitValidationSecurityFeature() {
+      this.securityBundleBuilder = SecurityBundle.builder().disableBufferLimitValidation();
+      return this;
+    }
+
+    @Override
+    public PlatformBundleBuilder<C> withCorsAllowedMethods(String... httpMethods) {
+      validateConfigureCors();
+      this.corsBundleBuilder.withAllowedMethods(httpMethods);
+      return this;
+    }
+
+    @Override
+    public PlatformBundleBuilder<C> withCorsAdditionalAllowedHeaders(
+        String... additionalAllowedHeaders) {
+      validateConfigureCors();
+      this.corsBundleBuilder.withAdditionalAllowedHeaders(additionalAllowedHeaders);
+      return this;
+    }
+
+    @Override
+    public PlatformBundleBuilder<C> withCorsAdditionalExposedHeaders(
+        String... additionalExposedHeaders) {
+      validateConfigureCors();
+      this.corsBundleBuilder.withAdditionalExposedHeaders(additionalExposedHeaders);
+      return this;
+    }
```

Comment:
Open for discussion: Do we prefer to do this in code or in config?

Reply:
I don't think that this is part of the deployment. This is part of the implementation. There shouldn't be any need to expose additional headers if the implementation doesn't require it. If you still want to override the default behavior of the implementation, use an ingress.

Reply:
Some services in our org use it to allow _"Content-Range"_ for a list result instead of providing paging parameters in a response model. Do you know other cases? This example does not match our Rest Guide, I think.

Reply:
`Content-Range` is not designed for paging as it requests specific bytes in a resource, but I won't judge this. My point is, that if the REST Api has such a feature, the REST API implementer has to allow the access to the header. We should let the consumer figure out that it doesn't work.

We had some implementations that expose the total count for paging as a header, but even there we have another recommendation in the REST guide.

Reply:
I thought that this was also part of the configuration class but it isn't. So no need for further actions.